tasklist/importer-860/src/main/java/io/camunda/tasklist/zeebeimport/v860/processors/es/VariableZeebeRecordProcessorElasticSearch.java

```diff
@@ -102,4 +109,120 @@ private UpdateRequest getVariableQuery(final VariableEntity entity) throws Persi
+    return variableListViewEntity;
+  }
+
+  private VariableListViewEntity associateVariableWithProcess(
+      final VariableEntity entity, final VariableListViewEntity variableListViewEntity) {
+    return associateVariableWithParent(
+        variableListViewEntity, "processVariable", entity.getProcessInstanceId());
+  }
+
+  private VariableListViewEntity associateVariableWithTask(
+      final VariableListViewEntity variableListViewEntity) {
+    return associateVariableWithParent(
+        variableListViewEntity, "taskVariable", variableListViewEntity.getScopeKey());
+  }
+
+  private VariableListViewEntity associateVariableWithParent(
+      final VariableListViewEntity variableListViewEntity,
+      final String name,
+      final String parentId) {
+    variableListViewEntity.getJoin().setName(name);
+    variableListViewEntity.getJoin().setParent(Long.valueOf(parentId));
+    return variableListViewEntity;
+  }
+
+  private boolean isProcessScope(final VariableEntity entity) {
+    return Objects.equals(entity.getProcessInstanceId(), entity.getScopeFlowNodeId());
+  }
+
+  private boolean isTaskOrSubProcessVariable(final VariableEntity entity) {
+    return !Objects.equals(entity.getProcessInstanceId(), entity.getScopeFlowNodeId());
+  }
+
+  private VariableEntity getVariableEntity(
+      final Record record, final VariableRecordValueImpl recordValue) {
+    final VariableEntity entity = new VariableEntity();
+    entity.setId(
+        VariableEntity.getIdBy(String.valueOf(recordValue.getScopeKey()), recordValue.getName()));
+    entity.setKey(record.getKey());
+    entity.setPartitionId(record.getPartitionId());
+    entity.setScopeFlowNodeId(String.valueOf(recordValue.getScopeKey()));
+    entity.setProcessInstanceId(String.valueOf(recordValue.getProcessInstanceKey()));
+    entity.setName(recordValue.getName());
+    entity.setTenantId(recordValue.getTenantId());
+    if (recordValue.getValue().length()
+        > tasklistProperties.getImporter().getVariableSizeThreshold()) {
+      // store preview
+      entity.setValue(
+          recordValue
+              .getValue()
+              .substring(0, tasklistProperties.getImporter().getVariableSizeThreshold()));
+      entity.setIsPreview(true);
+    } else {
+      entity.setValue(recordValue.getValue());
+    }
+    entity.setFullValue(recordValue.getValue());
+    return entity;
+  }
+
+  private UpdateRequest prepareUpdateRequest(
+      final VariableListViewEntity variableListViewEntity, final String routingKey)
+      throws PersistenceException {
+    try {
+      final UpdateRequest request =
+          new UpdateRequest()
+              .index(tasklistListViewTemplate.getFullQualifiedName())
+              .id(variableListViewEntity.getId())
+              .upsert(objectMapper.writeValueAsString(variableListViewEntity), XContentType.JSON)
+              .routing(routingKey)
+              .doc(objectMapper.writeValueAsString(variableListViewEntity), XContentType.JSON)
```

Comment:
for VariableEntity we update only value, fullValue and isPreview

Reply:
Yes, :) But similar to UserTask, where we can send the whole object to the Update Request, I  did the same over here, also works. Over this case, This method is used for CREATE/UPDATE, so can be both. 

If does not exist -> Create the Variable
If it exist -> It will update the existent object with the id